src/compiler/checker.ts

```diff
@@ -2161,19 +2161,46 @@ namespace ts {
                     }
                 }
             }
+
+            // The invalid initializer error is needed in two situation:
+            // 1. When result is undefined, after checking for a missing "this."
+            // 2. When result is defined
+            const checkAndReportErrorForInvalidInitializer = ()=>{

```

Comment:
```suggestion
            function checkAndReportErrorForInvalidInitializer() {
```

Is more the style we have in this codebase.